x/campaign/types/campaign.go

```diff
@@ -37,7 +39,7 @@ func (m Campaign) Validate() error {
 		return errors.New("invalid allocated shares")
 	}
 
-	if IsTotalSharesReached(m.AllocatedShares) {
+	if IsTotalSharesReached(m.AllocatedShares, spntypes.TotalShareNumber) {
```

Comment:
it's the same problematic here as in the previous comment
We should rely on the value set in the store

Since `Validate` is used for genesis validation, I would suggest adding `TotalShares` as part of the genesis state (not param though), the value can then be accessed in genesis `Validate` and this campaign `Validate` can take `TotalShare` as a parameter